### ISSUE TITLE
[FR] `thickness_scaling` and `scalefontsizes`

### ISSUE THREAD

**Issue by jolars** [JuliaPlots/Plots.jl#2811]
I think the behavior of `thickness_scaling` warrants a discussion. It is, at least to me, surprising that `thickness_scaling` results  in font scaling. I wonder if it might not be a better solution to separate the behavior of `thickness_scaling` and `scalefontsizes` such that they do not interfere with one another, i.e. have `thickness_scaling` only affect non-text elements. The current behavior seems unintuitive to me and makes it awkward to adjust only non-text elements, since it necessitates simultaneously scaling font sizes inversely. (Unless I am missing something?)

I understand the desire to want a one-argument solution to upscaling the entire plot, but might it not be better to in that case introduce a third setting that affects the entire plot?

**Comment by isentropic** [comment]
Perhaps it should be called 'plot scale' or 'layout scale' 
It is meant to optimize fonts for presentations and posters

**Comment by isentropic** [comment]
It is a quite rare case when you would want to just increase the thicknesses, correct me if I am wrong

**Comment by jolars** [comment]
> It is a quite rare case when you would want to just increase the thicknesses, correct me if I am wrong

Perhaps, but I can see situations in which this would come up. Consider plotting multiple plots in a single layout for instance. It is fairly common to want to reduce the size of the plots but not the font size in this instance. Reducing the thickness of lines and shapes in a plot doesn't hurt accessibility much whereas reducing font size does. I don't think I'm the only one who's experienced visualizations in papers for which the size of the plots have been reduced so much that they're hard read.

**Comment by isentropic** [comment]
Perhaps this is what you mean 
```
plot([plot(rand(10)) for i in 1:4]..., thickness_scaling=1)
plot([plot(rand(10)) for i in 1:4]..., thickness_scaling=2)

```

![image](https://user-images.githubusercontent.com/24591123/85700472-5f58b000-b717-11ea-9903-9df40b5d8216.png)
![image](https://user-images.githubusercontent.com/24591123/85700562-76979d80-b717-11ea-811b-7dd289661d20.png)

I think thickness_scaling is exactly what you would need and behaves very well for these sorts of needs

**Comment by jolars** [comment]
> Perhaps this is what you mean
> 
> ```
> plot([plot(rand(10)) for i in 1:4]..., thickness_scaling=1)
> plot([plot(rand(10)) for i in 1:4]..., thickness_scaling=2)
> ```
> 
> ![image](https://user-images.githubusercontent.com/24591123/85700472-5f58b000-b717-11ea-9903-9df40b5d8216.png)
> ![image](https://user-images.githubusercontent.com/24591123/85700562-76979d80-b717-11ea-811b-7dd289661d20.png)
> 
> I think thickness_scaling is exactly what you would need and behaves very well for these sorts of needs

No, that's exactly what I don't want. Or, rather, I think that it's great that it's possible to scale everything, including fonts, and I think it's very useful for presentations and posters. 

But I also think it would be good to also have the possibility to only scale the non-font elements and I think that it's generally more intuitive to have two arguments that don't overlap in functionality than two have two arguments where one does. To me, this behavior would be preferable:

```julia
plot(x, y, plot_scaling = 1.2, font_scaling = 1.2) # scales everything
plot(x, y, font_scaling = 1.2) # scales fonts
plot(x, y, plot_scaling = 1.2) # scales everything but fonts
```

Right now, however, to scale only non-fonts I have to do this (unless I am mistaken):

```julia
a = 0.7
Plots.scalefontsizes(1/a)
plot(x, y, thickness_scaling = a)
```

I don't think this is a reasonable interface (including the fact that `scalefontsizes` and `thickness_scaling` are set in completely different ways).

**Comment by daschw** [comment]
I think that's a good point @jolars
